src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -58,10 +58,9 @@ internal bool AddGas(long gas)
         protected override void ContextUnloaded(ExecutionContext context)
         {
             base.ContextUnloaded(context);
-            if (context.EvaluationStack == CurrentContext?.EvaluationStack) return;
-            int rvcount = context.GetState<ExecutionContextState>().RVCount;
-            if (rvcount != -1 && rvcount != context.EvaluationStack.Count)
-                throw new InvalidOperationException();
+            if (CurrentContext != null && context.EvaluationStack != CurrentContext.EvaluationStack)
+                if (context.EvaluationStack.Count == 0)
+                    Push(StackItem.Null);
```

Comment:
We can do this in the compiler, and here throw an exception?

Reply:
No, we can't. If the method is called by `CALL`, it shouldn't push anything. If the method is called by `System.Contract.Call`, it should push `null` onto the stack. So there is nothing to do with the compiler.

The VM will only automatically copy the parameters and return values when it is called across contracts (because the calls inside the contract share the stack). Therefore, the VM should automatically fill a `null` for the method that returns `void`.